unison-core/src/Unison/LabeledDependency.hs

```diff
@@ -10,47 +10,65 @@ module Unison.LabeledDependency
   , effectConstructor
   , fold
   , referents
-  , toReference
-  , LabeledDependency
+  , LabeledDependency(..)
+  , pattern ConstructorReference
+  , pattern TermReference
   , partition
   ) where
 
 import Unison.Prelude hiding (fold)
 
 import Unison.ConstructorType (ConstructorType(Data, Effect))
 import Unison.Reference (Reference(DerivedId), Id)
-import Unison.Referent (Referent, pattern Ref, pattern Con, ConstructorId)
+import Unison.Referent (Referent, ConstructorId)
 import qualified Data.Set as Set
+import qualified Unison.Referent as Referent
 
--- dumb constructor name is private
-newtype LabeledDependency = X (Either Reference Referent) deriving (Eq, Ord, Show)
+-- | A Union Type which contains either Type References or Term Referents.
+data LabeledDependency =
```

Comment:
I plan to rename this to `LabeledRef` since it's used for a lot more than dependencies now, but I'll do that separately.